modules/network/src/integrationTest/java/org/apache/ignite/internal/network/processor/ItTransferableObjectProcessorIncrementalTest.java

```diff
@@ -0,0 +1,339 @@
+import static org.apache.ignite.internal.network.processor.IncrementalCompilationConfig.readClassName;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.testing.compile.JavaFileObjects;
+import com.squareup.javapoet.ClassName;
+import java.io.BufferedReader;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import javax.tools.DiagnosticCollector;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaCompiler.CompilationTask;
+import javax.tools.JavaFileObject;
+import javax.tools.JavaFileObject.Kind;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+import javax.tools.ToolProvider;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Integration tests for the {@link TransferableObjectProcessor} incremental compilation.
+ */
+public class ItTransferableObjectProcessorIncrementalTest {
+    /**
+     * Package name of the test sources.
+     */
+    private static final String RESOURCE_PACKAGE_NAME = "org.apache.ignite.internal.network.processor";
+
+    /** File manager for incremental compilation. */
+    private InMemoryJavaFileManager fileManager;
+
+    /** Javac diagnostic collector. */
+    private DiagnosticCollector<JavaFileObject> diagnosticCollector = new DiagnosticCollector<>();
+
+    @BeforeEach
+    void setUp() {
+        JavaCompiler systemJavaCompiler = ToolProvider.getSystemJavaCompiler();
+        StandardJavaFileManager standardFileManager = systemJavaCompiler.getStandardFileManager(diagnosticCollector, Locale.getDefault(),
+                StandardCharsets.UTF_8);
+
+        this.fileManager = new InMemoryJavaFileManager(standardFileManager);
+    }
+
+    @Test
+    public void testIncrementalRemoveTransferable() throws Exception {
+        String testMessageGroup = "MsgGroup";
+        String testMessageGroupName = "GroupName";
+        String testMessageClass = "TestMessage";
+        String testMessageClass2 = "SomeMessage";
+
+        var compilationObjects1 = new ArrayList<JavaFileObject>();
+        JavaFileObject messageGroupObject = createMessageGroup(testMessageGroup, testMessageGroupName);
+        compilationObjects1.add(messageGroupObject);
+        compilationObjects1.add(createTransferable(testMessageClass, 0));
+
+        Map<URI, JavaFileObject> compilation1 = compile(compilationObjects1);
+
+        JavaFileObject messageRegistry1 = compilation1.get(uriForMessagesFile());
+        try (BufferedReader bufferedReader = new BufferedReader(messageRegistry1.openReader(true))) {
```

Comment:
I can't fully agree with this testing approach, since you are relying on the internal deserialization mechanism. I think that using mocks and `readConfig` might be a better solution, for example:
```
var env = mock(ProcessingEnvironment.class);
var filer = mock(Filer.class);

when(env.getFiler()).thenReturn(filer);
when(filer.getResource(StandardLocation.CLASS_OUTPUT, "", CONFIG_FILE_NAME))
        .thenReturn(messageRegistry1);

var config = IncrementalCompilationConfig.readConfig(env);
```
What do you think? This way you won't have to use internal `readClassName` methods

Reply:
Sounds legit